### PR TITLE
Feature: Empty state UI for LHN

### PR DESCRIPTION
**Pull request by tienifr**
### Details

This PR implements an "empty state" UI when users have cleared out the LHN in #focus mode on small screen devices.

### Fixed Issues

$ https://github.com/Expensify/App/issues/34926
PROPOSAL: https://github.com/Expensify/App/issues/34926#issuecomment-1905274451

### Tests

**Small screen devices only**

1. Switch to `#focus` mode
2. Mark all reports as read or unpin them so that LHN is cleared out
3. Verify the empty state UI shows as expected
4. Verify it in different themes, languages and devices

- [x] Verify that no errors appear in the JS console

### Offline tests

NA

### QA Steps

**Small screen devices only**

1. Switch to `#focus` mode
2. Mark all reports as read or unpin them so that LHN is cleared out
3. Verify the empty state UI shows as expected

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos

<details>
<summary>Native</summary>

https://github.com/Expensify/App/assets/113963320/c2c5598f-dcfe-4fc9-a18e-da09e60a1ed9

https://github.com/Expensify/App/assets/113963320/6f4be5b9-8fdf-4447-91c5-529f47a4fd0b

</details>

<details>
<summary>mWeb</summary>

https://github.com/Expensify/App/assets/113963320/230e7de6-b8d7-41c6-8148-63b937cc4527

https://github.com/Expensify/App/assets/113963320/4233f2f1-e677-40bb-a082-ba0731acb09d

https://github.com/Expensify/App/assets/113963320/254999ab-0a97-4b41-adb5-74375b0074b1

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@tienifr Yeah, ping me again when all logic is ready

**Comment by roryabraham** [comment]
design label didn't work so I'm manually requesting review from @dannymcclain, who created the mockups

**Comment by DylanDylann** [comment]
@tienifr Jest Unit Tests / Storybook tests (pull_request) is failed

File: src/components/LHNOptionsList/LHNOptionsList.tsx
Comment (DylanDylann) [inline src/components/LHNOptionsList/LHNOptionsList.tsx:63]:
```suggestion
    const renderEmptyLHNSubtitle = useCallback(
```

File: src/components/LHNOptionsList/LHNOptionsList.tsx
Comment (DylanDylann) [inline src/components/LHNOptionsList/LHNOptionsList.tsx:49]:
```suggestion
    const shouldShowEmptyLHN = isSmallScreenWidth && data.length === 0;
```

File: src/styles/variables.ts
Comment (DylanDylann) [inline src/styles/variables.ts:126]: Do we confirm the width and height of the icon?

File: src/components/LHNOptionsList/LHNOptionsList.tsx
Comment (DylanDylann) [inline src/components/LHNOptionsList/LHNOptionsList.tsx:167]: @tienifr Why do we need to add this prop?

File: src/components/BlockingViews/BlockingView.tsx
Comment (DylanDylann) [inline src/components/BlockingViews/BlockingView.tsx:39]: I think we can simply this type like this

```
type BlockingViewProps = RequiredIllustrationProps & {
    /** Animation for the page */
    animation?: DotLottieAnimation;

    /**
     * Expensicon for the page
     * If animation is provided, icon is not required
     */
    icon?: React.FC<SvgProps> | ImageSourcePropType;
```

File: src/components/BlockingViews/BlockingView.tsx
Comment (DylanDylann) [inline src/components/BlockingViews/BlockingView.tsx:113]: Should update to use `useCallback` hooks
```
const renderSubtitle = useCallback(
```

File: .storybook/webpack.config.js
Comment (DylanDylann) [inline .storybook/webpack.config.js:40]: @tienifr Please help to explain more detail why we need to add this change.

**Comment by dannymcclain** [comment]
@roryabraham thanks! I'm having a hard time getting the LHN to actually clear out—but the videos in the description look really good.

Any tips for getting rid of this error with the iOS ad hoc build? I feel like I can't get the LHN to clear out because I'm trying to do it on desktop.

Screenshot 2024-03-06 at 8 51 14 AM


cc @Expensify/design for vis.

**Comment by shawnborton** [comment]
Hmm I just tested the adhoc build and I am just getting a blank LHN:

IMG_8955

**Comment by roryabraham** [comment]
@dannymcclain can you create a GitHub issue and assign me? I'll need to add your device to the AdHoc provisioning profile (this should be made part of onboarding, if it's not already)

Not sure but you might be able to reproduce on desktop by making the screen narrow like a phone

File: src/styles/variables.ts
Comment (DylanDylann) [reply src/styles/variables.ts:126]: It seems `iconSizeSmall` looks good here

File: src/components/LHNOptionsList/LHNOptionsList.tsx
Comment (DylanDylann) [reply src/components/LHNOptionsList/LHNOptionsList.tsx:167]: Please help to detail why `px unit works differently between native and web platforms`

**Comment by DylanDylann** [comment]
@tienifr That makes sense to me. The logic to hide/display reports in `OptionRowLHN` should be only handled in `shouldReportBeInOptionList` function. Let's focus on https://github.com/Expensify/App/issues/37296 first

cc @roryabraham

**Comment by roryabraham** [comment]
ok, going to put this on HOLD for https://github.com/Expensify/App/issues/37296

**Comment by shawnborton** [comment]
Sure, will start a new build

**Comment by shawnborton** [comment]
Seems to work well on my end!

**Comment by roryabraham** [comment]
AdHoc iOS build failed with:

> Error: declaration of 'RCTConvert' must be imported from module 'React.RCTConvert' before it is required

I have seen this before and in the past it was a cache issue that meant we needed to run `pod install` but were not. I thought this was fixed though...

File: src/components/BlockingViews/BlockingView.tsx
Comment (roryabraham) [inline src/components/BlockingViews/BlockingView.tsx:17]: @tienifr can you clarify the expected behavior you're trying to convey with these types?

It _looks_ like you're trying to say something like "either icon or animation is required, but not both". Is that right?

Maybe what you want is more like this? ([source](https://github.com/sindresorhus/type-fest/blob/main/source/merge-exclusive.d.ts))

```ts
import type {MergeExclusive} from 'type-fest';

/**
 * This page requires either an icon or an animation, but not both
 */
type RequiredIllustrationProps = MergeExclusive<
    {
        /** Expensicon for the page */
        icon: React.FC<SvgProps> | ImageSourcePropType;
    },
    {
        /** Animation for the page */
        animation: DotLottieAnimation;
    }
>;
```

File: src/components/BlockingViews/BlockingView.tsx
Comment (roryabraham) [inline src/components/BlockingViews/BlockingView.tsx:68]: should `animationStyles` and `animationWebStyle` be grouped such that they can only be provided alongside the `animation` prop, but not with the `icon` prop?

File: src/components/BlockingViews/BlockingView.tsx
Comment (roryabraham) [inline src/components/BlockingViews/BlockingView.tsx:114]: this is just a bit different from many other files in the repo. Let's change both `renderContent` and `renderSubtitle` to just use `useMemo` and return the rendered content instead?

File: src/components/LHNOptionsList/LHNOptionsList.tsx
Comment (roryabraham) [inline src/components/LHNOptionsList/LHNOptionsList.tsx:69]: I think that `useMemo` is more appropriate than `useCallback` here and will simplify things

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Comment by DylanDylann** [comment]
@tienifr Please ping me when this PR is ready

**Review by dannymcclain — DISMISSED** [review]
From a design perspective I think this is looking and feeling good. Obviously make sure to still get a proper technical review from an engineer.

cc @Expensify/design in case y'all have any other feedback.

File: src/components/BlockingViews/BlockingView.tsx
Comment (roryabraham) [inline src/components/BlockingViews/BlockingView.tsx:89]: Let's keep it simple here and rely on type inference:

```suggestion
    const subtitleText = useMemo(
```

File: src/components/BlockingViews/BlockingView.tsx
Comment (roryabraham) [inline src/components/BlockingViews/BlockingView.tsx:109]:
```suggestion
    const subtitleContent = useMemo(() => {
```

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: src/components/BlockingViews/BlockingView.tsx
Comment (DylanDylann) [inline src/components/BlockingViews/BlockingView.tsx:38]: @tienifr Using camelCase convention

**Comment by DylanDylann** [comment]
https://github.com/Expensify/App/blob/b2826932eeada95171028c92470972cec08342f9/src/components/LHNOptionsList/OptionRowLHN.tsx#L50-L52

Should we remove this code to ensure that all logic to hide reports in LHN is moved to LHNOptionsList

File: .storybook/webpack.config.ts
Comment (DylanDylann) [reply .storybook/webpack.config.ts]: @tienifr If we don't add this change, which story test will be failed

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: .storybook/webpack.config.ts
Comment (DylanDylann) [reply .storybook/webpack.config.ts]: Oh yeah, It helps remove some warnings but I still see errors in Story Test. But it seems it isn't related to this PR.

Screenshot 2024-04-01 at 11 36 46

**Review by DylanDylann — APPROVED** [review]
@roryabraham Could you help to re-run all tests?

**Comment by DylanDylann** [comment]
@roryabraham  I approved this PR to re-run all tests. [A minor comment](https://github.com/Expensify/App/pull/37361#issuecomment-2029111631). Everything else looks good

File: src/components/BlockingViews/BlockingView.tsx
Comment (roryabraham) [reply src/components/BlockingViews/BlockingView.tsx:38]: `PascalCase` is correct for component props

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
not an emergency, we're working on fixing this: https://expensify.slack.com/archives/C07J32337/p1711999860875239?thread_ts=1711998323.452789&cid=C07J32337

**Pull request merged**
